### PR TITLE
Feature/Introduce PersonNameComponents for Google, Facebook, LinkedIn sign in

### PR DESCRIPTION
**Pull request by EgzonArifi**

### CODE REVIEW

File: Sources/Core/PersonNameComponents+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+public extension PersonNameComponents {
+  static func create(namePrefix: String? = .none,
+                     middleName: String? = .none,
+                     givenName: String? = .none,
+                     familyName: String? = .none,
+                     nameSuffix: String? = .none,
+                     nickname: String? = .none,
+                     phoneticRepresentation: PersonNameComponents? = .none) -> PersonNameComponents {
+    var components = PersonNameComponents()
```
Comment (borut-t) [inline Sources/Core/PersonNameComponents+Extension.swift:31]: Re-indent using new line for each param please.

**Review by borut-t — APPROVED** [review]
Awesome 🙌

File: Sources/Core/PersonNameComponents+Extension.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+//
+//  Created by Egzon Arifi on 09/11/2024.
+//
+
+import AuthenticationServices
+import Foundation
+
+public extension PersonNameComponents {
+  var name: String? {
+    guard let givenName = givenName else {
```
Comment (yllfejziu) [inline Sources/Core/PersonNameComponents+Extension.swift:13]: Let's use new syntax: `guard let givenName {}`

File: Sources/Core/PersonNameComponents+Extension.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      return familyName
+    }
+    guard let familyName = familyName else {
```
Comment (yllfejziu) [inline Sources/Core/PersonNameComponents+Extension.swift:16]: Let's use new syntax here too: `guard let familyName {}`

File: Sources/Core/PersonNameComponents+Extension.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      return givenName
+    }
+    return "\(givenName) \(familyName)"
+  }
+}
+
+public extension PersonNameComponents {
+  static func create(
```
Comment (yllfejziu) [inline Sources/Core/PersonNameComponents+Extension.swift:24]: I'd suggest we create an `init` instead of a static func.

**Review by yllfejziu — APPROVED** [review]
Looks good 👌 
Just a few nitpicks below.

File: Sources/Core/PersonNameComponents+Extension.swift [hunk 5]
```diff
@@ -10,34 +10,31 @@ import Foundation
...
-    components.nickname = nickname
-    components.phoneticRepresentation = phoneticRepresentation
-    return components
+  init(namePrefix: String? = .none,
+       middleName: String? = .none,
+       givenName: String? = .none,
+       familyName: String? = .none,
+       nameSuffix: String? = .none,
+       nickname: String? = .none,
+       phoneticRepresentation: PersonNameComponents? = .none) {
```
Comment (borut-t) [inline Sources/Core/PersonNameComponents+Extension.swift:30]: Let's put every param into a new line.

**Review by borut-t — APPROVED** [review]

**Pull request merged**